nucliadb/src/nucliadb/search/search/rerankers.py

```diff
@@ -187,3 +145,58 @@ def get_reranker(kind: search_models.Reranker) -> Reranker:
         logger.warning(f"Unknown reranker requested: {kind}. Using multi-match booster instead")
         reranker = MultiMatchBoosterReranker()
     return reranker
+
+
+def apply_reranking(results: KnowledgeboxFindResults, reranked: list[RankedItem]):
+    inverted_results = {}
+    for rid, resource in results.resources.items():
+        for field_id, field in resource.fields.items():
+            for paragraph_id, paragraph in field.paragraphs.items():
+                inverted_results[paragraph_id] = (
+                    paragraph,
+                    (field_id, field),
+                    (rid, resource),
+                )
+
+    # order results by new score
+    best_matches = []
```

Comment:
should this be a set to handle duplicated paragraphs (same paragraph matching different indexes)?

Reply:
Sure, but if you don't mind, I'll do it in another PR and fix the one in find_merge too